proofs/tree.go

```diff
@@ -446,12 +460,11 @@ func (n *LeafNode) HashNode(h hash.Hash) error {
 }
 
 // ConcatValues concatenates property, value & salt into one byte slice.
-func ConcatValues(prop string, value string, salt []byte) (payload []byte, err error) {
-	propBytes := []byte(prop)
+func ConcatValues(propBytes []byte, value string, salt []byte) (payload []byte, err error) {
 	payload = append(payload, propBytes...)
 	payload = append(payload, []byte(value)...)
 	if len(salt) != 32 {
-		return []byte{}, fmt.Errorf("%s: Salt has incorrect length: %d instead of 32", prop, len(salt))
+		return []byte{}, fmt.Errorf("%s: Salt has incorrect length: %d instead of 32", propBytes, len(salt))
```

Comment:
This error isn't really human readable anymore if we use bytes but I guess there's no easy way to avoid it.

Reply:
I didn't know whether it was better to leave it or remove it.

Reply:
Yeah, let's leave it unless go has a built in check to see if a type `[]byte` is a valid ASCII string. Then we could do an if/else but not worth adding any code just for this.

Reply:
I have refactored ConcatValues to take a PropertyName. PropertyNames can be encoded as []byte or printed in a human-readable fashion. This leads to a nicely formatted error messages here in either case. Let me know what you think.

Reply:
Awesome!